### ISSUE TITLE
Fail to integrate WSL2 in docker desktop version 4.5

### ISSUE THREAD

**Issue by felizbuild** [docker/for-win#12544]
Hi all, 
There are 4 problems in version 4.5 while it works fine in 4.4.4.

1. The docker cannot start until I use the command   "C:\Program Files\Docker\Docker\DockerCli.exe" -SwitchDaemon
2. even started docker engine by  "C:\Program Files\Docker\Docker\DockerCli.exe" -SwitchDaemon, 
    under the general settings:

   "Use the WSL 2 based engine (Windows Home can only run the WSL 2 backend)" is greyed out **without tick**
   (However, it is ticked in 4.4.4)

3. Under resources settings,
    only proxy is shown, not WSL integration is present.

4. docker ps command in WSL2 should error like cannot find socket

Info:
- Windows Version:             11 version 21H2 (OS Build 22000.434) upgraded from win10 Home
- Docker Desktop Version:   4.5
- distribution:                       Unbuntu(18.04) LTS / Unbuntu(20.04)   [both are tried] 
- backend:                            WSL2  (Hyper-V is not available on windows Home)

**Comment by mccaa25** [comment]
I am having same exact problems with Docker 4.5.0(74594).
Windows 10 build 19044
Ubunutu 20.04 LTS

**Comment by mccaa25** [comment]
I went to release notes for Docker Desktop for Windows and downloaded 4.4.2, uninstalled 4.5.0 and installed 4.4.2 as administrator (not sure if this was necessary), and now it works fine. Spent 6 hours trying to get 4.5.0 to work. Clearly there is a bug in 4.5.0!

**Comment by mccaa25** [comment]
** one point to note though, when I run 4.4.2 it does say there is an update to 4.4.4 so it might be worth grabbing this one if you want to roll back to an older one. I haven't tried updating to this but I think the problem occurs with 4.5, so 4.4.4 might be safe. 4.4.2 definitely works for me.

**Comment by mccaa25** [comment]
**Ok well, it "tricked" me into downloading 4.5.0 because it said it was going to update to 4.4.4,
However this time, 4.5.0 worked immediately. The WSL integration box is grayed out but this time it is checked! Before it was unchecked.

**Comment by code-grey** [comment]
> I am having same exact problems with Docker 4.5.0(74594).
> Windows 10 build 19044
> Ubunutu 20.04 LTS

Same here, now about to try out Docker 4.4.2

**Comment by code-grey** [comment]
> ** one point to note though, when I run 4.4.2 it does say there is an update to 4.4.4 so it might be worth grabbing this one if you want to roll back to an older one. I haven't tried updating to this but I think the problem occurs with 4.5, so 4.4.4 might be safe. 4.4.2 definitely works for me.

4.4.4 didn't work for me though, like i downloaded 4.4.4 and that very day 4.5.0 update rolled out

**Comment by mccaa25** [comment]
You could try 4.4.2. I downloaded this and then updated to 4.5 through the
docker desktop dashboard and it still works!
Or (and maybe an easier solution)...
In another issue thread a solution was posted that seems to work: Just go
to the config file at C:\Users<username>\AppData\Roaming\Docker\settings.json,
and set "wslEngineEnabled": true that will fix this issue for v4.5 :)

**Comment by mat007** [comment]
Hi, thanks for reporting, and sorry about the inconvenience, I’ll close this issue in favor of #12545 which has a temporary [workaround](https://github.com/docker/for-win/issues/12545#issuecomment-1037225140) until we release Desktop 4.5.1
